client/hotrod-client/src/main/java/org/infinispan/client/hotrod/configuration/ConfigurationChildBuilder.java

```diff
@@ -49,9 +50,18 @@
    /**
     * For replicated (vs distributed) Hot Rod server clusters, the client balances requests to the
     * servers according to this strategy.
+    *
+    * @deprecated since 9.3, use {@link #balancingStrategy(Supplier)} instead.
     */
+   @Deprecated
    ConfigurationBuilder balancingStrategy(FailoverRequestBalancingStrategy balancingStrategy);
```

Comment:
This will break users as is since they'd expect to pass in a `org.infinispan.client.hotrod.impl.transport.tcp.FailoverRequestBalancingStrategy` here. This method should still take org.infinispan.client.hotrod.impl.transport.tcp.FailoverRequestBalancingStrategy, as fully qualified name, and then we could have a wrapper that delegates to the newer location.

Reply:
@galderz `org.infinispan.client.hotrod.impl.transport.tcp.FailoverRequestBalancingStrategy` implements `org.infinispan.client.hotrod.FailoverRequestBalancingStrategy`, so this can be still safely passed in. Recompilation is needed as the method signature has changed but that's an allowed change (we don't guarantee binary compatibility).